File: services/server/database/env.go
package database

import (
    "fmt"
    "os"
)

type Environment struct {
    address  string
    username string
    password string
    dbname   string
}

func (env *Environment) dsn() string {
    return fmt.Sprintf(
        "%s:%s@tcp(%s)/%s?charset=utf8mb4&parseTime=True&loc=Local",
        env.username, env.password, env.address, env.dbname)
}

// loadEnv load environments from OS to Environment struct
func loadEnv() *Environment {
    return &Environment{
        address:  os.Getenv("MYSQL_ADDRESS"),
        username: os.Getenv("MYSQL_USERNAME"),
        password: os.Getenv("MYSQL_PASSWORD"),
        dbname:   os.Getenv("MYSQL_DBNAME"),
    }
}

File: services/collector/src/collector/database/mysql/env.py
import os
from dataclasses import dataclass

from dotenv import load_dotenv


@dataclass
class Environments:
    """Database environments."""

    user: str
    password: str
    host: str
    port: int
    dbname: str

    def url(self) -> str:
        """Provide the Database URL."""
        address = f"{self.host}:{self.port}" if self.port else self.host
        return f"mysql+asyncmy://{self.user}:{self.password}@{address}/{self.dbname}"

    @classmethod
    def parse(cls) -> "Environments":
        """Create a DB environment object from the environment file."""
        load_dotenv()
        address = os.getenv("MYSQL_ADDRESS")
        host, port = address.split(":") if ":" in address else (address, "0")
        return cls(
            user=os.getenv("MYSQL_USERNAME"),
            password=os.getenv("MYSQL_PASSWORD"),
            host=host,
            port=int(port),
            dbname=os.getenv("MYSQL_DBNAME")
        )
